htdocs/core/actions_massactions.inc.php

```diff
@@ -488,6 +488,20 @@
 
 					complete_substitutions_array($substitutionarray, $langs, $objecttmp, $parameters);
 
+					$contactarr = $objecttmp->liste_contact(-1, 'external', 0, '', 1);
```

Comment:
$objecttmp may be empty.
Also method liste_contact may not exists.

Reply:
I have taken your suggestion into account.
If there is 2 different for the same type/code, then we use the first one.